### PR TITLE
pts-stage1: scripts for benchmarking perf.

### PR DESCRIPTION
**Pull request by solaslin**
1. git_info.py and spec.py are prepared in advance for pts website backend. 
2. calling `./hipblastlt-perf` can produce the csv for our current interested problem sets (first version in this PR)
3. This PR shows how to add benchmark set for matmul (using client.cpp) and amax (using client_extop_amax.cpp)
4. A quick example is to run `clients/scripts/performance/hipblaslt-perf --suite all` and then you can get csv files

### CODE REVIEW

File: clients/benchmarks/client_extop_amax.cpp [hunk 1]
```diff
@@ -238,7 +238,11 @@ int AmaxTest(hipDataType type, hipDataType dtype, int m, int n, hipblaslt_initia
 
     float dur{};
     hipErr = hipEventElapsedTime(&dur, beg, end);
-    std::cout << "Time elapsed: " << std::to_string(dur / numRuns) << " ms\n";
+    dur *= 1000; // to mju-s
+
+    std::cout << "[amax]:m,n,type,dtype,us\n";
+    std::cout << std::to_string(m) << "," << std::to_string(n) << "," << hip_datatype_to_string(type) << ","
+              << hip_datatype_to_string(dtype) << "," << std::to_string(dur / numRuns) << std::endl;
```
Comment (solaslin) [inline clients/benchmarks/client_extop_amax.cpp:246]: output something for bench script to capture

File: clients/scripts/performance/problems/matmul_probset1_bench.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  #   - { transA: T, transB: N }
+  #   - { transA: T, transB: T }
+
+Tests:
+- name: matmul_f8f16_f16
+  function:
+    matmul: *fp8fp16_precision_dst_fp16
+  matrix_size:
+    - { M:  16,    N:  16,    K:  1024  }
+    - { M:  16,    N:  16,    K:  8192  }
```
Comment (solaslin) [inline clients/scripts/performance/problems/matmul_probset1_bench.yaml:18]: later, we can add any routine test problem sizes as a new yaml file

File: clients/scripts/performance/suites.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    problemlist = [Problem(args={"--yaml" : "matmul_probset1_bench.yaml"})]
+    yield ProblemSet(benchType="matmul", name="benchset_1", problems=problemlist)
+
+
+def all():
+    """all routine benchmarks"""
+
+    yield from amax_set_1()
+    yield from matmul_set_1()
```
Comment (solaslin) [inline clients/scripts/performance/suites.py:93]: A handy way to add/define new "problem-sets"

File: clients/scripts/performance/bench.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2021 - 2022 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (jichangjichang) [inline clients/scripts/performance/bench.py]: 2024

Comment (solaslin) [reply clients/scripts/performance/bench.py]: done

**Comment by solaslin** [comment]
Has fixed more Copyright year in .py files

**Review by KKyang — APPROVED** [review]

**Review by hcman2 — APPROVED** [review]

**Pull request merged**